tests/db_functions/datetime/test_extract_trunc.py

```diff
@@ -1044,6 +1044,24 @@ def test_trunc_timezone_applied_before_truncation(self):
         self.assertEqual(model.melb_year.year, 2016)
         self.assertEqual(model.pacific_year.year, 2015)
 
+    def test_trunc_timezone_with_nonexistent_or_ambiguous_result(self):
+        """
+        If truncating to a nonexistent or ambiguous datetime, assume normal time (is_dst=False).
+        """
+        sao = pytz.timezone('America/Sao_Paulo')
+        start_datetime = sao.localize(datetime(2016, 10, 16))  # Nonexistent time
+        end_datetime = sao.localize(datetime(2016, 2, 20, 23))  # Ambiguous time
+        self.create_model(start_datetime, end_datetime)
+
+        with timezone.override(sao):
+            model = DTModel.objects.annotate(
+                truncated_start=TruncDay('start_datetime'),
+                truncated_end=TruncHour('end_datetime'),
+            ).get()
+
+        self.assertEqual(model.truncated_start.dst(), timedelta(0))
+        self.assertEqual(end_datetime.dst(), timedelta(0))
```

Comment:
Pretty sure you want to test `model.end_datetime` here?

Reply:
true, good catch!